app/routes/questionnaire.py

```diff
@@ -89,6 +93,19 @@ def get_questionnaire(schema, questionnaire_store):
         questionnaire_store.metadata,
     )
 
+    if request.method == "POST":
+        if not schema.is_hub_enabled():
+            raise NotFound
+
+        if router.is_survey_complete():
+            submission_handler = SubmissionHandler(
+                schema, questionnaire_store, router.full_routing_path()
+            )
+            submission_handler.submit_questionnaire()
+            return redirect(url_for("post_submission.get_thank_you"))
+
+        return redirect(router.get_first_incomplete_location_in_survey_url())
+
     if not router.can_access_hub():
         redirect_location_url = router.get_first_incomplete_location_in_survey_url()
         return redirect(redirect_location_url)
```

Comment:
The `is_hub_enabled` check in the POST handling should be removed, and `router.can_access_hub` moved from after the POST handling to be the first check.

```suggestion
    if not router.can_access_hub():
        redirect_location_url = router.get_first_incomplete_location_in_survey_url()
        return redirect(redirect_location_url)
        
    if request.method == "POST":
        if router.is_survey_complete():
            submission_handler = SubmissionHandler(
                schema, questionnaire_store, router.full_routing_path()
            )
            submission_handler.submit_questionnaire()
            return redirect(url_for("post_submission.get_thank_you"))

        return redirect(router.get_first_incomplete_location_in_survey_url())
```

Reply:
It looks like the suggested change for this was made and then reverted?

Reply:
@ajmaddaford was your suggestion with the intention to also go to first location even if hub is not enabled?. Which is what it will do, even though previously it would 404.

Reply:
Yes. It seemed odd (and unnecessary) for the POST to behave differently from the GET.